crates/sui-bridge/src/config.rs

```diff
@@ -0,0 +1,267 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::crypto::BridgeAuthorityKeyPair;
+use crate::eth_client::EthClient;
+use crate::sui_client::SuiClient;
+use anyhow::anyhow;
+use ethers::types::Address as EthAddress;
+use fastcrypto::traits::EncodeDecodeBase64;
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
+use std::collections::BTreeMap;
+use std::path::PathBuf;
+use std::str::FromStr;
+use std::sync::Arc;
+use sui_config::Config;
+use sui_sdk::SuiClient as SuiSdkClient;
+use sui_types::base_types::ObjectRef;
+use sui_types::base_types::{ObjectID, SuiAddress};
+use sui_types::crypto::SuiKeyPair;
+use sui_types::digests::TransactionDigest;
+use sui_types::object::Owner;
+use sui_types::Identifier;
+use tracing::info;
+
+#[serde_as]
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct BridgeNodeConfig {
+    /// The port that the server listens on.
+    pub server_listen_port: u16,
+    /// The port that for metrics server.
+    pub metrics_port: u16,
+    /// Path of the file where bridge authority key (Secp256k1) is stored as Base64 encoded `privkey`.
+    pub bridge_authority_key_path_base64_raw: PathBuf,
+    /// Rpc url for Sui fullnode, used for query stuff and submit transactions.
+    pub sui_rpc_url: String,
+    /// Rpc url for Eth fullnode, used for query stuff.
+    pub eth_rpc_url: String,
+    /// Path of the file where bridge client key (any SuiKeyPair) is stored as Base64 encoded `flag || privkey`.
+    /// If `run_client` is true, and this is None, then use `bridge_authority_key_path_base64_raw` as client key.
+    pub bridge_client_key_path_base64_sui_key: Option<PathBuf>,
+    /// Whether to run client. If true, `bridge_client_key_path_base64_sui_key`,
+    /// `bridge_client_gas_object` and `db_path` needs to be provided.
+    pub run_client: bool,
+    /// The gas object to use for paying for gas fees for the client. It needs to
+    /// be owned by the address associated with bridge client key.
+    pub bridge_client_gas_object: Option<ObjectID>,
```

Comment:
I have not looked into the client implementation, but a shot in the dark here, would providing a pool of gas objects to be used be beneficial?

Reply:
yeah that'd be an optimization if concurrency is a concern. But itself is not trivial to implement, so keep it simple here